examples/siro_sandbox/app_states/app_state_fetch.py

```diff
@@ -4,132 +4,437 @@
 # This source code is licensed under the MIT license found in the
 # LICENSE file in the root directory of this source tree.
 
+from typing import Final
+
 import magnum as mn
 import numpy as np
 from app_states.app_state_abc import AppState
 from camera_helper import CameraHelper
 from controllers.gui_controller import GuiHumanoidController
+from gui_avatar_switch_helper import GuiAvatarSwitchHelper
 from gui_navigation_helper import GuiNavigationHelper
+from gui_pick_helper import GuiPickHelper
+from gui_throw_helper import GuiThrowHelper
 from utils.gui.gui_input import GuiInput
 from utils.gui.text_drawer import TextOnScreenAlignment
 from utils.hablab_utils import (
     get_agent_art_obj_transform,
     get_grasped_objects_idxs,
 )
 
+from habitat.datasets.rearrange.navmesh_utils import get_largest_island_index
```

Comment:
Can you explain this file at a high level? I was guessing it would resemble the latest AppStateFetch from the P0 demo branch, but when I diff against that file, I see this is missing lots of functionality related to Spot.

If this is some kind of reduced-functionality version of AppStateFetch, then I would suggest we call it something else instead of `app_state_fetch.py`.

Reply:
Related note to reviewers: I would not focus on the differences in this file compared to `main`'s `app_state_fetch.py`. As I understand, Mikael authored this starting with [this file](https://github.com/facebookresearch/habitat-lab/blob/eundersander/SIRo_hitl_Sep5_demo_refactored/examples/siro_sandbox/app_states/app_state_fetch.py) and removing Spot functionality.

Reply:
This is the `AppStateFetch` application from the P0 project. Because the Fetch state machine is not compatible with Habitat 3.0, and because we don't need it, this functionality was removed.

The result is that the Spot robot uses the policy from `social_rearrange/pop_play.yaml` - it does not intelligently react to the human's actions. However, the new features that were added in P0 are working (remote control, VR, grabbing, throwing, teleporting to navmesh, etc.).

Here's what changed:
* All references and code related to `fetch_baselines_controller` was removed (`FetchState,  `FetchBaselinesController`)
* Hard-coded initial agent positions and spawning logic were removed. Additionally, there was code to make sure that the robot and human spawn near each other.
* The resulting dead functions were removed:
    * _make_agent_face_target()
    * _make_robot_near_human()
    * _fetch_object_remote()
    * _get_target_object_bounding_box()
    * _draw_box_in_pos()
    * _get_state_machine_agent_translation()
    * _get_spot_status_text()
    * _viz_fetcher()
    * _update_place_goal()
    * _update_fetcher()

>If this is some kind of reduced-functionality version of AppStateFetch, then I would suggest we call it something else instead of app_state_fetch.py.

Agreed. "Fetch" is a thing of the past at this point.

Reply:
Why is the state machine not compatible with Habitat 3.0? I believe that the `pop_play.yaml` policy won't be compatible with a pick-throw demo because that high level policy expects a limited set of objects and placing locations. I understand that the previous code was very specific to P0 but being able to use a state machine or something more flexible than a policy seems valuable.